pmd-java/src/main/resources/rulesets/java/unnecessary.xml

```diff
@@ -13,7 +13,6 @@ The Unnecessary Ruleset contains a collection of rules for unnecessary code.
     <rule ref="category/java/errorprone.xml/UnusedNullCheckInEquals" deprecated="true" />
     <rule ref="category/java/errorprone.xml/UselessOperationOnImmutable" deprecated="true" />
 
-    <rule ref="category/java/codestyle.xml/UnnecessaryFinalModifier" deprecated="true" />
```

Comment:
I'm uncertain, whether we should provide the rule additionally under the old name:

    <rule ref="category/java/codestyle.xml/UnnecessaryModifier" name="UnnecessaryFinalModifier" depreated="true" />

That way, if someone uses "java-unnecessary/UnnecessaryFinalModifier", we would still resolve the rule. It could lead to duplicated reports - the rule might be in use twice.

 But on the other hand, the RuleSetFactoryCompatibility class should take care of this...

WDYT?

Reply:
I believe the RuleSetFactoryCompatibility already handles that scenario.

The only difference AFAIK is on the documentation site.

I think leaving it there threw an exception upon parsing the dogfooding ruleset, but I can't remember right now.

Reply:
No, it doesn't fail. I may have forgotten the `name` attribute 'though. I'm adding it back.